src/websocket/test/test_websockets_server.py

```diff
@@ -24,149 +23,195 @@
-    assert server.host == "localhost"
-    assert server.port == 8080
-    print(server.gpu_config)
+    server = WebSocketServer(config=MOCK_CONFIG)
     assert server.gpu_config == MOCK_CONFIG["websocket_stream"]["cuda"]
     assert server.cpu_config == MOCK_CONFIG["websocket_stream"]["cpu"]
 
 
 @patch("src.websocket.stream_transcriber.Transcriber.for_cpu", return_value=Mock())
 def test_websocket_server_transcribers(mock_for_cpu):
-    server = WebSocketServer("localhost", 8080, MOCK_CONFIG)
+    server = WebSocketServer(config=MOCK_CONFIG)
     mock_for_cpu.assert_called_once_with(
-       model_name="tiny", cpu_threads=1, num_workers=1
+        model_name="tiny", cpu_threads=1, num_workers=1
     )
     assert server.cpu_transcriber is not None
 
 
-@pytest.mark.asyncio
-async def test_start_stream_and_response_to_text():
-    server = WebSocketServer("localhost", 8080, MOCK_CONFIG)
-
-    # Run the server in another thread, as it is blocking
-    server_thread = threading.Thread(
-        target=asyncio.run, args=(run_server_in_thread(server),)
-    )
-    server_thread.start()
-
-    await asyncio.sleep(0.5)  # Give the server some time to start
-
-    try:
-        async with websockets.connect("ws://localhost:8080") as websocket:
-            await websocket.send("Hello")
-            response = await websocket.recv()
-            assert response == "control message unknown"
-            await websocket.close()
-    finally:
-        server.stop_server()
-        server_thread.join(timeout=1)
-
-
-@pytest.mark.asyncio
-async def test_send_eof_and_expect_connection_close():
-    server = WebSocketServer("localhost", 8080, MOCK_CONFIG)
-
-    # Run the server in another thread, as it is blocking
-    server_thread = threading.Thread(
-        target=asyncio.run, args=(run_server_in_thread(server),)
-    )
-    server_thread.start()
-
-    await asyncio.sleep(0.5)  # Give the server some time to start
-
-    response: str = "" # We need to get the export files with the server response
-    try:
-        async with websockets.connect("ws://localhost:8080") as websocket:
-            print("WebSocket connection established, sending 'eof'")
-            await websocket.send("eof")
-
-            # We expect the server to close the connection after receiving 'eof'
+def test_start_stream_and_response_to_text():
+    client = TestClient(app)
+
+    with client.websocket_connect("/") as websocket:
+        websocket.send_text("Hello")
+        response = websocket.receive_text()
+        assert response == "control message unknown"
+
+
+def test_send_eof_and_expect_connection_close():
+    client = TestClient(app)
+    with client.websocket_connect("/") as websocket:
+        websocket.send_text("eof")
+        try:
+            # TODO: how to test the connection is then closed?
+            # Ends with a string response, else it would be a json response
+            # TODO: can come an empty string? Should be properly validated what comes back?
```

Comment:
Do this! Just copy the validation in the very last test.

assert len(message) == 32